### PR TITLE
Replace Gitter and IRC by a Bytecode Alliance Zulip stream link

### PR DESCRIPTION
**Pull request by bnjbvr**
The Bytecode Alliance will be using [Zulip](https://bytecodealliance.zulipchat.com/#narrow/stream/217117-cranelift/topic/general) for instant communication. There is a cranelift stream there, and a topic for general discussions. If you don't know about zulip, think of a stream as a collection of discussions around a general topic; then there's the ability to create specific topics within a given stream. I've added links in the README and CONTRIBUTING files of this repository, as a starting point.

[README rendered](https://github.com/bnjbvr/cranelift/blob/chat-room/README.md)

Till or Dan, can you have a look, please? Thanks!

### CODE REVIEW

**Comment by tschneidereit** [comment]
I think it might be better to link to just the stream, not a specific topic: we don't want to encourage people to habitually use the `general` topic instead of more specific topics as appropriate. With that: LGTM!

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -13,13 +13,11 @@ is reflected in the code or documentation yet. If you see things that seem
...
 expect them to, we're interested to hear about it!
 
-We have a [chat room on Gitter], and questions are also welcome as issues
-in the [Cranelift issue tracker]. Some folks also hang out in the #cranelift
-IRC channel on [irc.mozilla.org].
+We have a [stream] in the Bytecode Alliance Zulip instance, and questions are
+also welcome as issues in the [Cranelift issue tracker].
 
-[chat room on Gitter]: https://gitter.im/CraneStation/Lobby
+[stream]: https://bytecodealliance.zulipchat.com/#narrow/stream/217117-cranelift/topic/general
```
Comment (bjorn3) [inline CONTRIBUTING.md]:
```suggestion
[stream]: https://bytecodealliance.zulipchat.com/#narrow/stream/217117-cranelift
```

File: README.md [hunk 2]
```diff
@@ -12,7 +12,7 @@ into executable machine code.
 [![Documentation Status](https://readthedocs.org/projects/cranelift/badge/?version=latest)](https://cranelift.readthedocs.io/en/latest/?badge=latest)
 [![Build Status](https://github.com/bytecodealliance/cranelift/workflows/CI/badge.svg)](https://github.com/bytecodealliance/cranelift/actions)
 [![Fuzzit Status](https://app.fuzzit.dev/badge?org_id=bytecodealliance)](https://app.fuzzit.dev/orgs/bytecodealliance/dashboard)
-[![Gitter chat](https://badges.gitter.im/bytecodealliance/bytecodealliance.svg)](https://gitter.im/CraneStation/Lobby)
+[![Chat](https://img.shields.io/badge/chat-zulip-brightgreen.svg)](https://bytecodealliance.zulipchat.com/#narrow/stream/217117-cranelift/topic/general)
```
Comment (bjorn3) [inline README.md]:
```suggestion
[![Chat](https://img.shields.io/badge/chat-zulip-brightgreen.svg)](https://bytecodealliance.zulipchat.com/#narrow/stream/217117-cranelift)
```

**Comment by bnjbvr** [comment]
Thanks @bjorn3 for the suggestions! We actually had a chat with Till, and there's some likelihood (in my opinion) that people would be confused by the UI if landing on a stream, and would start asking random questions under random topics. No strong opinion here, just intuition, so we can start with what's done here and adjust if it doesn't work out!

**Comment by bjorn3** [comment]
:+1:

**Pull request merged**